doc/configuration.rst

```diff
@@ -18,12 +18,94 @@ The name can be seem from ``doit help`` output::
    -f ARG, --file=ARG        load task from dodo FILE [default: dodo.py]  (config: dodoFile)
 
 
+pyproject.toml
+--------------
+
+`doit` configuration can be read from `pyproject.toml <https://www.python.org/dev/peps/pep-0518/>`_
+under the `tool.doit` namespace. In the future, especially if a TOML parser becomes
+part of the python standard library, this will become the preferred configuration source,
+and may gain features not available in the legacy `doit.cfg`.
+
+.. note::
+
+   As a TOML parser is _not_ yet part of the standard library, a third-party package is
+   required, one of:
+
+   - `toml <https://pypi.org/project/toml/>`_
+   - `tomlkit <https://pypi.org/project/tomlkit/>`_
+
+
+TOML vs INI
+^^^^^^^^^^^
+
+While mostly similar, `TOML <https://toml.io>`_ differs from the INI format
+in a few ways:
+
+- all strings must be quoted with `'` or `"`
+- triple-quoted strings may contain new line characters (`\n`) and quotes
+- must be saved as UTF-8
+- integers and floating point numbers can be written without quotes
+- boolean values can be written unquoted and lower-cased, as `true` and `false`
+
+Unlike "plain" TOML, `doit` will parse pythonic strings into their correct types,
+e.g. `"True"`, `"False"`, `"3"`, but using "native" TOML types may be preferable.
+
+
+tool.doit
+^^^^^^^^^
+
+The `tool.doit` section may contain command line options that will be used
+(if applicable) by any commands.
+
+Example setting the DB backend type:
+
+.. code-block:: toml
+
+   [tool.doit]
+   backend = "json"
+
+All commands that have a `backend` option (*run*, *clean*, *forget*, etc),
+will use this option without the need for this option in the command line.
+
+
+tools.doit.commands
+^^^^^^^^^^^^^^^^^^^
+
+To configure options for a specific command, use a section with
+the command name under `tools.doit.commands`:
+
+.. code-block:: toml
+
+   [tools.doit.commands.list]
+   status = true
+   subtasks = true
+
+
+tools.doit.plugins
+^^^^^^^^^^^^^^^^^^
+
+Check the :ref:`plugins <plugins>` section for an introduction
+on available plugin categories.
+
+
+tools.doit.tasks
+^^^^^^^^^^^^^^^^
+
+To configure options for a specific task, use a section with
+the task name under `tool.doit.tasks`:
+
+.. code-block:: toml
+
+   [tool.doit.tasks.make_cookies]
+   cookie_type = "chocolate"
+   temp = "375F"
+   duration = 12
 
```

Comment:
You mentioned about being able to define tasks in TOML, right?

Have you thought about what the name would be? I would thing `doit.tasks`. So maybe get a different name for setting task parameters `doit.params.<task_name>`, what do you think?